algorithms/indexing/lattice_search_strategies.py

```diff
@@ -0,0 +1,611 @@
+                if r_dist > sqrt(sq_relp_band + sq_candidate_band):
+                    bad_candidate = True
+                    break
+            if bad_candidate:
+                continue
+
+            # Calculate co-planarity of the relps, including the origin
+            points = flex.vec3_double(exp_relps + [cand_vec, (0.0, 0.0, 0.0)])
+            plane = least_squares_plane(points)
+            plane_score = flex.sum_sq(
+                points.dot(plane.normal) - plane.distance_to_origin
+            )
+
+            # Reject if the group of relps are too far from lying in a single plane.
+            # This cut-off was determined by trial and error using simulated images.
+            if plane_score > 6e-7:
+                continue
+
+            # Construct a graph including the accepted candidate node
+            g = graph.factory_add_vertex(
+                {
+                    "spot_id": cand["spot_id"],
+                    "miller_index": cand["miller_index"],
+                    "rlp_datum": cand["rlp_datum"],
+                },
+                weights_to_other=residual_dist,
+            )
+
+            result.append(g)
+
+        return result
+
+    @staticmethod
+    def _fit_U_from_superposed_points(reference, other):
+
+        # Add the origin to both sets of points
+        origin = flex.vec3_double(1)
+        reference.extend(origin)
+        other.extend(origin)
+
+        # Find U matrix that takes ideal relps to the reference
+        fit = superpose.least_squares_fit(reference, other)
+        return fit.r
+
+    def _fit_crystal_model(self, graph):
+
+        vertices = graph.vertices
+
+        # Reciprocal lattice points of the observations
+        sel = flex.size_t([e["spot_id"] for e in vertices])
+        reference = self.spots["rlp"].select(sel)
+
+        # Ideal relps from the known cell
+        other = flex.vec3_double([e["rlp_datum"] for e in vertices])
+
+        U = self._fit_U_from_superposed_points(reference, other)
+        UB = U * self.Bmat
+
+        if self._params.bootstrap_crystal:
+
+            # Attempt to index the low resolution spots
+            from dials_algorithms_indexing_ext import AssignIndices
+
+            phi = self.spots["xyzobs.mm.value"].parts()[2]
+            UB_matrices = flex.mat3_double([UB])
+            result = AssignIndices(self.spots["rlp"], phi, UB_matrices, tolerance=0.3)
+            hkl = result.miller_indices()
+            sel = hkl != (0, 0, 0)
+            hkl_vec = hkl.as_vec3_double().select(sel)
+
+            # Use the result to get a new UB matrix
+            reference = self.spots["rlp"].select(sel)
+            other = self.Bmat.elems * hkl_vec
+            U = self._fit_U_from_superposed_points(reference, other)
+            UB = U * self.Bmat
+
+        # Calculate RMSD of the fit
+        other_rotated = U.elems * other
+        rms = reference.rms_difference(other_rotated)
```

Comment:
Again, is there anything to be gained by assigning `U.elems * other` to a separate variable only to use it in the following line?